nntrainer/layers/cl_layers/rmsnorm_layer_cl.cpp

```diff
@@ -0,0 +1,398 @@
+    ret = inputbuf.WriteData(context.command_queue_inst_, data);
+    if (!ret) {
+      break;
+    }
+
+    ret = gammabuf.WriteData(context.command_queue_inst_, gdata);
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      0, &inputbuf, sizeof(cl_mem));
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      1, &resultbuf, sizeof(cl_mem));
+    if (!ret) {
+      break;
+    }
+
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      2, &gammabuf, sizeof(cl_mem));
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      4, &b, sizeof(int));
+    if (!ret) {
+      break;
+    }
+
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      3, &epsilon, sizeof(cl_half));
+    if (!ret) {
+      break;
+    }
+
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      5, &c, sizeof(int));
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      6, &h, sizeof(int));
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm_fp16.SetKernelArguments(
+      7, &w, sizeof(int));
+    if (!ret) {
+      break;
+    }
+    const int work_groups_count[3] = {(b*h*c), 1, 1};
+    const int work_group_size[3] = {32, 32, 1}; // test-value
+
+    ret = context.command_queue_inst_.DispatchCommand(
+      RMSNormLayerCl::kernel_rmsnorm_fp16, work_groups_count, work_group_size);
+    if (!ret) {
+      break;
+    }
+
+    ret = resultbuf.ReadData(context.command_queue_inst_, rdata);
+    if (!ret) {
+      break;
+    }
+
+  } while (false);
+
+}
+
+void RMSNormLayerCl::incremental_forwarding(nntrainer::RunLayerContext &context,
+                                          unsigned int from, unsigned int to,
+                                          bool training) {
+  Tensor &in = context.getInput(SINGLE_INOUT_IDX);
+  Tensor &out = context.getOutput(SINGLE_INOUT_IDX);
+  Tensor &gamma = context.getWeight(wt_idx[RMSParams::gamma]);
+  ml::train::TensorDim in_dim = in.getDim();
+  ml::train::TensorDim out_dim = out.getDim();
+
+  ml::train::TensorDim in_step_dim = in_dim;
+  ml::train::TensorDim out_step_dim = out_dim;
+
+  if (from) {
+    NNTR_THROW_IF(to - from != 1, std::invalid_argument)
+      << "incremental step size is not 1";
+    from = 0;
+    to = 1;
+  }
+
+  in_step_dim.height(to - from);
+  out_step_dim.height(to - from);
+
+  Tensor in_step = in.getSharedDataTensor(in_step_dim, 0, true);
+  Tensor out_step = out.getSharedDataTensor(out_step_dim, 0, true);
+
+  auto &epsilon = std::get<props::Epsilon>(rmsnorm_props).get();
+
+  if (in_step.getDataType() == ml::train::TensorDim::DataType::FP32) {
+          rmsnormProcess(in,out,gamma,epsilon,context);
+  }
+  else{
+      rmsnormProcess_fp16(in,out,gamma,epsilon,context);
+  }
```

Comment:
inconsistent format.

```suggestion
  if (in_step.getDataType() == ml::train::TensorDim::DataType::FP32) {
    rmsnormProcess(in, out, gamma, epsilon, context);
  } else {
    rmsnormProcess_fp16(in, out, gamma, epsilon, context);
  }
```